tools/scripts/refresh-ubuntu-amis.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+"""
+Fetch the latest official Ubuntu AMIs that we hardcode into our system.
+
+This will update all *_master_ami and *_bastion_ami tags in bumpenvs.yaml file.
+
+Usage: refresh-ubuntu-amis.py path/to/bumpenvs.yaml
+"""
+
+import sys
+from typing import Dict, List
+
+import requests
+import yaml
+
+get_cache = {}  # type: Dict[str, bytes]
+
+
+def cacheable_get(url: str) -> bytes:
+    global get_cache
+    if url not in get_cache:
+        req = requests.get(url)
+        assert req.ok, f"request to {url} failed with HTTP {req.status_code}"
```

Comment:
non-blocking: `req.raise_for_status()`, perhaps.

Reply:
Ah, I looked for that function but couldn't find it and assumed I was think of some other library : )

Reply:
done